modules/home/software/karabiner/default.nix

```diff
@@ -2,8 +2,12 @@
 
 with lib;
 
-{
-  config = mkIf (config.shabka.darwinConfig != {}) {
+let
+  cfg = config.shabka.karabiner;
+in {
+  options.shabka.karabiner.enable = mkEnableOption "Enable karabiner (only relevant for Darwin hosts).";
+
+  config = mkIf (cfg.enable && (config.shabka.darwinConfig != {})) {
```

Comment:
I'm thinking for this, we should probably remove the `config.shabka.darwinConfig != {}` check now. It was here because we did not have an enable.

Reply:
Why not just leave it? Karabiner shouldn't be built on linux or nixos hosts

Reply:
I see your point. In that case, I would do an assertion instead so if the user does enable it, it throws an error instead of silently fail.